### PR TITLE
reset password functionality

### PR DESCRIPTION
**Pull request by j2kan**
To test: add your email associated with taform locally (I've done this on the server for the team gmail already).

1. Follow the read me for testing locally.

Changed some urls around to be more consistent and cleaner.

added password reset and password change functionality
added 404 not found page so any mistyped url will see this 404 page that can lead the user back to the ta form
added favicon
added timeout of session after 1 hour

### CODE REVIEW

**Review by smwatts — COMMENTED** [review]
Lgtm!

**Review by lauradobson — CHANGES_REQUESTED** [review]
- Tested using: !@#QWEasdzx as the new password. Still worked although the language said "password must not be too similar to the old password". Would remove this line if you're not doing a check for that.


- "Your password can't be entirely numeric." what is a commonly used password? I would give examples or remove this line.

- I know you're using Django admin css but can you use the CSS we're using to make the system look more unified?

- Logged out of the system, then logged into django admin. I went to test home.html and i was logged into the system. I guess this is supposed to happen?

- For the waterloo favicon can you find a version that has a transparent background instead of white? Log into learn and look at there icon as an example. This is a small change and I'm okay if this doesn't get done, but I think it looks awkward.

- "Got lost? Apply to be a TA. TA Application" Trying to access password reset when logged out. This should say something more along the lines of "Got lost? You must be signed in to access this page" then redirect them to the login page. This is currently confusing the roles of TA and AC.

-In the email "Your username, in case you've forgotten: taform" bold the username

- Server error when I click on the link in the email
image

**Comment by j2kan** [comment]
@lauradobson 
Tested using: !@#QWEasdzx as the new password. Still worked although the language said "password must not be too similar to the old password". Would remove this line if you're not doing a check for that.

I think the language says
`Your password can't be too similar to your other personal information.`, there's nothing that stops you from changing it to the same password.

I won't be removing any of the other statements:
Your password can't be too similar to your other personal information.
Your password must contain at least 8 characters.
Your password can't be a commonly used password.
Your password can't be entirely numeric.

Because they will be held true in django's validation. For example a password of '12345678' won't work.

I actually prefer the admin style css and would leave it for the all the password functionality since it's all django related. Our buttons such as upload and submit still need css, we only have css for the home and logout button atm.

Yep logged into admin = logged into the system but we never expose /admin page and i will remove it from the urls actaully. 

New favicon updated sorry just screenshotted a random waterloo image last night without cropping.

I will also add a login link to the 404 page for the AC and keep the existing thing for the student.

Thanks for taking the time to test! Also the 500 error should be fixed now.

**Comment by j2kan** [comment]
screen shot 2018-02-20 at 10 12 12 am

new cropped.

**Comment by lauradobson** [comment]
image
didn't see the icon change come through

**Comment by lauradobson** [comment]
other than that it looks great!

File: taworks/taform/templates/accounts/password_reset_email.html [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+
+Please go to the following page and choose a new password:
+{% block reset_link %}
+{{ protocol }}://{{ domain }}{% url 'password_reset_confirm' uidb64=uid token=token %}
+{% endblock %}
+Your username, in case you've forgotten:{{ user.get_username }}
+
+Thanks for using our site!
+
+Ta Works
```
Comment (amyleblond) [inline taworks/taform/templates/accounts/password_reset_email.html]: -- for consistently sake, I think we are 'TAWorks' ?? @smwatts this was our final choice right lol?

File: taworks/taform/templates/admin/base_site.html [hunk 2]
```diff
@@ -1,9 +1,5 @@
 {% extends "admin/base.html" %}
-
-{% block title %}{{ title }} | {{ site_title|default:_('Django site admin') }}{% endblock %}
-
+<link rel="shortcut icon" href="{{ STATIC_URL }}/static/favicon/favicon.ico">
 {% block branding %}
-<h1 id="site-name"><a href="{% url 'admin:index' %}">TA Administration</a></h1>
+<h1 id="site-name">TA Works</h1>
```
Comment (amyleblond) [inline taworks/taform/templates/admin/base_site.html]: same here

**Review by amyleblond — DISMISSED** [review]
really neat -- just two naming things for TAWorks -- I think this is what we officially chose - jsut for consistency.

File: taworks/taform/templates/accounts/password_reset_email.html (hunk 1 above)
Comment (smwatts) [reply taworks/taform/templates/accounts/password_reset_email.html]: screen shot 2018-02-20 at 2 31 48 pm

from the brochure material :)

File: taworks/taform/templates/admin/base_site.html (hunk 2 above)
Comment (j2kan) [reply taworks/taform/templates/admin/base_site.html]: Should i do lowercase W?

**Comment by j2kan** [comment]
@lauradobson i think you have to remove your cache for the new icon to come thorugh but i'm going to crop it some more

**Review by lauradobson — DISMISSED** [review]

**Review by smwatts — APPROVED** [review]
lgtm!

**Pull request merged**